File: solutions/python/0011.py
class Solution:
    def maxArea(height: list[int]) -> int:
        left = 0
        right = len(height) - 1
        max_water = 0

        while left < right:
            width = right - left
            h = min(height[left], height[right])
            current_water = width * h

            max_water = max(max_water, current_water)

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return max_water

File: solutions/go/0011.go
func maxArea(height []int) int {
    left := 0
    right := len(height) - 1
    maxWater := 0

    for left < right {
        width := right - left
        h := min(height[left], height[right])
        currentWater := width * h
        maxWater = max(maxWater, currentWater)

        if height[left] < height[right] {
            left++
        } else {
            right--
        }
    }

    return maxWater
}